### PR TITLE
[conductor] Remove PublishChannel and use MPA command

### PR DESCRIPTION
**Pull request by CaseyHillers**
Move more of the playbook into conductor. The MPA command inputs are prone to human error.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

### CODE REVIEW

File: dev/conductor/core/lib/src/globals.dart [hunk 1]
```diff
@@ -18,7 +18,7 @@ const List<String> kReleaseChannels = <String>[...kBaseReleaseChannels, Framewor
 
 const String kReleaseDocumentationUrl = 'https://github.com/flutter/flutter/wiki/Flutter-Cherrypick-Process';
 
-const String kLuciPackagingConsoleLink = 'https://ci.chromium.org/p/flutter/g/packaging/console';
+const String kLuciPackagingConsoleLink = 'https://ci.chromium.org/p/dart-internal/g/flutter_packaging/console';
```
Comment (christopherfujino) [inline dev/conductor/core/lib/src/globals.dart:21]: nice! I can never find this lol

File: dev/conductor/core/lib/src/next.dart [hunk 2]
```diff
@@ -142,14 +142,11 @@ class NextContext extends Context {
 
         await pushWorkingBranch(engine, state.engine);
       case pb.ReleasePhase.CODESIGN_ENGINE_BINARIES:
-        stdio.printStatus(<String>[
-          'You must validate pre-submit CI for your engine PR, merge it, and codesign',
-          'binaries before proceeding.\n',
```
Comment (christopherfujino) [inline dev/conductor/core/lib/src/next.dart:147]: So is MPA when codesigning happens now?

File: dev/conductor/core/lib/src/next.dart [hunk 3]
```diff
@@ -250,83 +247,18 @@ class NextContext extends Context {
...
-        await framework.pushRef(
-            fromRef: headRevision,
-            toRef: state.releaseChannel,
-            remote: state.framework.upstream.url,
-            force: force,
-        );
+        final String command = '''
+          tool-proxy-cli --tool_proxy=/abns/dart-eng-tool-proxy/prod-dart-eng-tool-proxy-tool-proxy.annealed-tool-proxy \\
+          --block_on_mpa -I flutter_release \\
+          :git_branch ${state.framework.candidateBranch} \\
```
Comment (christopherfujino) [inline dev/conductor/core/lib/src/next.dart:253]: this tool uses colons in front of CLI options?

File: dev/conductor/core/lib/src/proto/conductor_state.proto [hunk 4]
```diff
@@ -11,16 +11,16 @@ message Remote {
 enum ReleasePhase {
   // Release was started with `conductor start` and repositories cloned.
   APPLY_ENGINE_CHERRYPICKS = 0;
+
+  // Verify engine CI is green before opening framework PR.
   CODESIGN_ENGINE_BINARIES = 1;
```
Comment (christopherfujino) [inline dev/conductor/core/lib/src/proto/conductor_state.proto:16]: Should we just rename this? Although we persist this proto information to disk, it's really only intended to be read from disk while you're in the middle of a release.

File: dev/conductor/core/lib/src/state.dart [hunk 5]
```diff
@@ -286,11 +284,21 @@ String githubAccount(String remoteUrl) {
...
-    throw globals.ConductorException('There is no next ReleasePhase!');
+  switch (currentPhase) {
+    case ReleasePhase.PUBLISH_VERSION:
+      return ReleasePhase.VERIFY_RELEASE;
+    case ReleasePhase.APPLY_ENGINE_CHERRYPICKS:
+    case ReleasePhase.CODESIGN_ENGINE_BINARIES:
+    case ReleasePhase.APPLY_FRAMEWORK_CHERRYPICKS:
+    case ReleasePhase.VERIFY_RELEASE:
+    case ReleasePhase.RELEASE_COMPLETED:
+     final ReleasePhase? nextPhase = ReleasePhase.valueOf(currentPhase.value + 1);
```
Comment (christopherfujino) [inline dev/conductor/core/lib/src/state.dart:295]:
```suggestion
      final ReleasePhase? nextPhase = ReleasePhase.valueOf(currentPhase.value + 1);
```

File: dev/conductor/core/lib/src/state.dart [hunk 6]
```diff
@@ -286,11 +284,21 @@ String githubAccount(String remoteUrl) {
...
+      if (nextPhase != null) {
+        return nextPhase;
+      }
   }
-  return nextPhase;
+    throw globals.ConductorException('There is no next ReleasePhase!');
```
Comment (christopherfujino) [inline dev/conductor/core/lib/src/state.dart:300]:
```suggestion
  throw globals.ConductorException('There is no next ReleasePhase!');
```

**Review by christopherfujino — APPROVED** [review]
Some nits, but LGTM

File: dev/conductor/core/lib/src/proto/conductor_state.proto (hunk 4 above)
Comment (CaseyHillers) [reply dev/conductor/core/lib/src/proto/conductor_state.proto]: Done! Renamed to VERIFY_ENGINE_CI

File: dev/conductor/core/lib/src/next.dart (hunk 3 above)
Comment (CaseyHillers) [reply dev/conductor/core/lib/src/next.dart:253]: Yep! 😢

File: dev/conductor/core/lib/src/next.dart (hunk 2 above)
Comment (CaseyHillers) [reply dev/conductor/core/lib/src/next.dart:147]: No, codesigning happens on engine postsubmit, and it's verified with a framework presubmit test.

Comment (christopherfujino) [reply dev/conductor/core/lib/src/next.dart:147]: ahh, so the builders are like channel aware, and then flip the codesign bit when it knows it's a release?

Comment (CaseyHillers) [reply dev/conductor/core/lib/src/next.dart:147]: Yeah! Anything that is built from dart-internal's secure builders will be code signed, which our RC branches use.

Comment (christopherfujino) [reply dev/conductor/core/lib/src/next.dart:147]: ahh, interesting. so master channel engine builds use the luci.flutter.prod pool, but RC branches use dart-internal?

Comment (CaseyHillers) [reply dev/conductor/core/lib/src/next.dart:147]: Yep! Longer term, luci.flutter.prod will converge with dart-internal to build all engine postsubmit artifacts. Then anything that could be used will be code signed.

Comment (christopherfujino) [reply dev/conductor/core/lib/src/next.dart:147]: Very cool

**Pull request merged**